packages/mobile/src/identity/verification.ts

```diff
@@ -123,7 +128,13 @@ export function* fetchVerificationState(forceUnlockAccount?: boolean) {
       // we want to reset password before force unlock account
       clearPasswordCaches()
     }
-    yield call(unlockAccount, account, !!forceUnlockAccount)
+    const result: UnlockResult = yield call(unlockAccount, account, !!forceUnlockAccount)
+    if (result !== UnlockResult.SUCCESS) {
+      // This navigateBack has no effect if part of onboarding and returns to home or
+      // settings page if the user pressed on the back button when prompted for the PIN.
+      navigateBack()
```

Comment:
Why it has no effect when part of the onboarding?

Reply:
When part of the onboarding there are two cases, and I see now that I missed an edge case:

1- This screen shows up just after setting the PIN, so the PIN is not asked again after that and there is no risk of pressing the back button. However, if a user waits a few minutes between setting the PIN and starting the verification process, the account might have been blocked so the PIN is requested again and the user would be redirected to the "Confirm PIN screen", which is unwanted. User shouldn't be able to navigate back from this screen (there is no back button but they can swipe back on iOS). I'll fix this.
2- This is the case I had previously tested, if the user kills the app when on the verification education screen, when they open it again they'll see the same screen again. This is where the `navigateBack` has no effect, since there is no screen to navigate back to.

Reply:
Pushed a fix to disallow going back from this screen, which was the original wanted behaviour